crates/uv-pep440/src/version_specifier.rs

```diff
@@ -893,6 +888,90 @@ pub(crate) fn parse_version_specifiers(
     Ok(version_ranges)
 }
 
+/// A simple `~=` version specifier with a major, minor and (optional) patch version, e.g., `~=3.13`
+/// or `~=3.13.0`.
+#[derive(Clone, Debug)]
+pub struct TildeVersionSpecifier<'a> {
+    inner: Cow<'a, VersionSpecifier>,
```

Comment:
Alas I wrote this as using a borrowed version specifier then needed `bounding_specifiers` after `with_patch` and needed to reconstruct it to own the specifier. It's not a big deal, but that's why it is this way.